### PR TITLE
Add onclick handler to update game state

### PR DESCRIPTION
**Pull request by Narigo**
## What type of PR is this? (check all applicable)

- [x] Feature
- [x] Bug Fix
- [ ] Documentation Update
- [ ] Refactor
- [x] Optimization

## Description

I've added an on:click handler for the columns to update the game state. There were a few minor things to check (Array needed to initialize with `undefined` and we need to check the cell for `undefined` or it would see `0` player turn as falsy in the template as well.

## Related Tickets & Documents

This is an addition to what we worked on on the stream.

### UI accessibility concerns?

Definitely the styling of the pieces could be different and there is no possibility to focus the columns through a keyboard yet. Maybe we could style the columns as buttons to get that for free?

## Added/updated tests?

- [ ] Yes
- [x] No, and this is why: This is an addition to a work-in-progress branch
- [ ] I need help with writing tests

## [optional] What gif best describes this PR or how it makes you feel?

Give me more stuff to do

Also, for the checklists in the PR:
I'm thinking too much

### CODE REVIEW

File: src/lib/stores/gameState.ts [hunk 1]
```diff
@@ -13,5 +13,5 @@ interface GameState {
 export const gameState = writable<GameState>({
 	players: [{ color: 'R' }, { color: 'Y' }],
 	playerTurn: 1,
-	boardState: new Array(7).fill(new Array(6)),
+	boardState: new Array(7).fill(new Array(6).fill(undefined)),
```
Comment (nickytonline) [inline src/lib/stores/gameState.ts:16]: I guess the `.fill(undefined)` is so there are actual array elements initialized with t he value `undefined` or is this not the same as omitting the fill?

**Review by nickytonline — COMMENTED** [review]
This looks great @Narigo!

Just to keep things separated and so you don't need to work off of a branch instead of main, I'll put up a PR with the board and coin components with their rudimentary look, and once that's merged, you can base your PR off of `main`.

File: src/lib/stores/gameState.ts (hunk 1 above)
Comment (nickytonline) [reply src/lib/stores/gameState.ts:16]: The initial game state we have here would make sense as a separate PR based off of main. Just trying to slice up things as the initial branch I had was a WIP

File: src/lib/Board.svelte [hunk 2]
```diff
@@ -1,14 +1,42 @@
...
 	{#each $gameState.boardState as column, colIndex}
-		<div data-column={colIndex}>
+		<div
+			data-column={colIndex}
+			on:click={insertCoinAsCurrentPlayerIntoColumn(colIndex)}
+		>
 			{#each column as cell, rowIndex}
 				<div data-cell="{colIndex},{rowIndex}">
-					{#if cell}
+					{#if cell !== undefined}
```
Comment (nickytonline) [inline src/lib/Board.svelte:39]: The explicit check is so that we don't run into an issue with `0` being considered falsy I imagine?

**Comment by nickytonline** [comment]
@Narigo, no rush, but when you have a chance, mind updating the branch? There’s some conflicts.

File: src/lib/Board.svelte (hunk 2 above)
Comment (Narigo) [reply src/lib/Board.svelte:39]: Yes, if it checks only for falsy, we get into trouble for this `0` values

File: src/lib/stores/gameState.ts (hunk 1 above)
Comment (Narigo) [reply src/lib/stores/gameState.ts:16]: If an Array is not filled, the spaces in there are just `empty`, which seems to be another value than `null` or `undefined`. Or it skips over these values when for-looping maybe. Honestly, I didn't look too deep into it, but I remember that this hit me before when not initializing values and the `cell !== undefined` check did not work without this 😅

**Comment by Narigo** [comment]
@nickytonline conflict should be fixed now - but is the `board-and-pieces-wip` already merged or do you want to merge this into it first and then continue?

**Comment by nickytonline** [comment]
I won't be merging the board wip. We can go with what you have.

**Comment by nickytonline** [comment]
@Narigo if you base the branch off of main I can merge this.

**Review by nickytonline — APPROVED** [review]
Thanks for the PR @Narigo! 🚀

**Pull request merged**